File: python/problem455.py
class Solution:
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        if not s:
            print(0)
        g.sort()
        s.sort()

        su=0

        for i in range(len(g)):

            while  s:
                if g[i]<=s[0]:
                    su+=1
                    s.pop(0)
                    break
                else:
                    s.pop(0)
        return su

File: cpp/problem455.cpp
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int n_satisfied = 0;
        int ig = 0, is = 0, ng = g.size(), ns = s.size();
        while( ig < ng && is < ns ){
            while( g[ig] > s[is] && is < ns ) ++is;
            if( is < ns ){ ++is; ++ig; ++n_satisfied; }
        }
        return n_satisfied;
    }
};
